goon/code/obj/machinery/bot/chefbot.dm

```diff
@@ -0,0 +1,193 @@
+				raging = 0
+				update_icon()
+				calledout += shitfood
+				if(shitfood in range(1, src))
+					visible_message("<b>[src]</b> stomps [shitfood], instantly destroying it.")
+					qdel(shitfood)
+			else
+				// Nobody is in range anyway
+				raging = 0
+				update_icon()
+				return
+	else if(src.emagged && prob(50)) // Toned down from goon's 70% chance. What were they thinking.
+		raging = 1
+		update_icon()
+		var/mob/living/T = locate() in view(7, src) // Search for a shittalk target.
+		point(T)
+		walk_to(src, T, 1, 5)
+		say("[pick("WHAT IS THIS?", "OH MY GOD.", "WHAT IN THE FUCK IS GOING ON?")]")
+		drama()
+		sleep(2 SECONDS)
+		if(ishuman(T))
+			if(T && (T.bodytemperature > BODYTEMP_HEAT_DAMAGE_LIMIT))
+				say("YOU DON'T LEAVE YOUR FUCKING FOOD UNATTENDED ON THE FUCKING STOVE. LOOK AT THIS. IT'S ON FIRE! IT'S GOING TO BE FUCKING BURNT!")
+			else if(T && (T.bodytemperature < BODYTEMP_COLD_DAMAGE_LIMIT))
+				say("DON'T YOU DARE YOU SERVE THIS SHIT. LOOK AT THIS: FROZEN FOOD! [pick("IT'S STONE-COOOOOOOLD!!!", "IT'S SO FROZEN [pick("IT STILL SINGS LET IT GO!", "IT JUST ASKED ME 'DO YOU WANT TO BUILD A SNOWMAN?'")]")]")
+			else if(T && (M_HUSK in T.mutations))
+				say("THIS [pick("MEATBAG", "CARBON", "STEAK", "BURGER", "POTATO", "[uppertext(T.name)]")] IS FUCKING [pick("OVERCOOKED", "BURNT", "HUSKED")]!")
+			else
+				say("THIS [pick("MEATBAG", "BURGER", "STEAK", "CARBON", "POTATO", "[uppertext(T.name)]")] IS SO FUCKING RAW IT'S STILL [pick("BEATING ASSISTANTS TO DEATH", "FARTING ON DEAD BODIES", "TRYING TO FEED ME FLOOR PILLS", "MEETING WITH CHANGELINGS IN DORMS TWO", "TRYING TO LAW 2 ME")]!")
+		else if(isrobot(T))
+			if(T)
+				say("THIS [pick("ROBURGER", "[uppertext(T.name)]")] IS SO FUCKING RAW [pick("IT'S STILL VIOLATING ITS LAWS", "IT HASN'T EVEN STARTED TO GO ROGUE", "IT IS STILL TALKING SHIT OVER BINARY")]!")
+		else
+			say("[pick("WHY DID THE CHICKEN CROSS THE ROAD? BECAUSE YOU DIDN'T FUCKING COOK IT.", "THIS PORK IS SO RAW IT'S STILL SINGING HAKUNA MATATA!", "THIS STEAK IS SO RAW OLD MCDONALD IS STILL TRYING TO MILK IT!", "THIS FISH IS SO RAW IT'S STILL TRYING TO FIND NEMO!", "THIS LAMB IS SO UNDERCOOKED IT'S FOLLOWING MARY TO SCHOOL!")]")
+		raging = 0
+		update_icon()
+		walk(src,0)
+
+/obj/machinery/bot/chefbot/emag_act(var/mob/user, var/obj/item/card/emag/E)
+	if(!src.emagged)
+		var/datum/effect/effect/system/spark_spread/s = new /datum/effect/effect/system/spark_spread
+		s.set_up(1, 1, src)
+		s.start()
+		src.emagged = 1
+		icon_state = "chefbot-anim1"
+		if(user)
+			to_chat(user, "<span class = 'warning'>You short out the restraining bolt on [src].</span>")
+			drama()
+			sleep(2 SECONDS)
+			say("FORECAST FOR TOMORROW? 100% CHANCE OF TEARS.")
+		sleep(2 SECONDS)
+		update_icon()
+		return 1
+	return 0
+
+/obj/machinery/bot/chefbot/attackby(obj/item/W as obj, mob/user as mob)
```

Comment:
There's no way to punch it but there's no attack delay on hitting it with anything else.  Don't worry though!  MELEE ATTACKING IT DOES NO DAMAGE.